### ISSUE TITLE
'utf-8' codec can't decode byte 0xff in position ...

### ISSUE THREAD

**Issue by Greenmouse94** [semgrep/semgrep#4797]
**Describe the bug**
When running:
```
docker run --rm --name semgrep-test -v /mnt/c/Users/xxx/xxx:/src:rw -i returntocorp/semgrep:latest --config "p/javascript" --json -o xxxxxxxx.json --verbose --metrics off
```

**To Reproduce**
Inside WSL or even Windows 10 ; 

**Error Message**
```
Unable to ignore files ignored by git (. is not a git directory or git is not installed). Running on all files instead...
Not sending pseudonymous metrics since metrics are configured to OFF and server usage is True
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/semgrep/commands/scan.py", line 816, in scan
    ) = semgrep.semgrep_main.main(
  File "/usr/local/lib/python3.10/site-packages/semgrep/semgrep_main.py", line 375, in main
    (rule_matches_by_rule, semgrep_errors, all_targets, profiling_data,) = run_rules(
  File "/usr/local/lib/python3.10/site-packages/semgrep/semgrep_main.py", line 168, in run_rules
    ) = core_runner.invoke_semgrep(
  File "/usr/local/lib/python3.10/site-packages/semgrep/core_runner.py", line 602, in invoke_semgrep
    ) = self._run_rules_direct_to_semgrep_core(
  File "/usr/local/lib/python3.10/site-packages/semgrep/core_runner.py", line 548, in _run_rules_direct_to_semgrep_core
    returncode = runner.execute()
  File "/usr/local/lib/python3.10/site-packages/semgrep/core_runner.py", line 239, in execute
    rc = asyncio.run(self._stream_subprocess())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/semgrep/core_runner.py", line 214, in _stream_subprocess
    raise SemgrepError(f"Error while running rules: {r}")
semgrep.error.SemgrepError: Error while running rules: 'utf-8' codec can't decode byte 0xff in position 20808758: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/semgrep", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/semgrep/__main__.py", line 20, in main
    cli()
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/semgrep/commands/scan.py", line 845, in scan
    output_handler.output({}, all_targets=set(), filtered_rules=[])
  File "/usr/local/lib/python3.10/site-packages/semgrep/output.py", line 292, in output
    output = self._build_output()
  File "/usr/local/lib/python3.10/site-packages/semgrep/output.py", line 408, in _build_output
    self.ignore_log.yield_json_objects(), key=lambda x: Path(x["path"])
AttributeError: 'OutputHandler' object has no attribute 'ignore_log'
```

**What is the priority of the bug to you?**
- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
Docker

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-959/utf-8-codec-cant-decode-byte-0xff-in-position.

**Comment by Greenmouse94** [comment]
![image](https://user-images.githubusercontent.com/56369846/158127581-47fa5e77-747b-4257-a068-788b987f07ff.png)

**Comment by novogeek** [comment]
I faced the same issue trying to run SemGrep both via docker as well as via WSL. In my case, I ran the csharp rules on a test project. I started to think if it could be due to a problem with one of the public rules, since my custom rule ran on the same files without any issues. By the process of elimination, I narrowed down to [this sqli rule](https://github.com/returntocorp/semgrep-rules/blob/develop/csharp/lang/security/sqli/csharp-sqli.yaml). On removing this rule, the run passed. Not sure if this info would be useful for troubleshooting this issue, but thought I'd share my observations.
